New post: "Conan and Conan Center: 2021 in numbers" (#211)
conan-io/conan-io.github.io: _posts/2022-01-04-conan-stats-2021.markdown

---
layout: post
comments: false
title: "Conan and Conan Center: 2021 in numbers"
meta_title: "Conan and Conan Center: 2021 in numbers"
description: "Numbers and charts showing the state of Conan and Conan Center in 2021"
meta_description: "Numbers and charts showing the state of Conan and Conan Center in 2021"
---

2021 has been quite a busy year for Conan. We are working hard to get Conan 2.0 ready, almost 4
years after 1.0 was released. And, of course, the 1.X series is still very active, with 10
versions released during the year.

And Conan Center has not lagged behind. In 2021, only two years after its inception, we

> celebrated with our community a great milestone: on August, Conan Center

Reviewer: ```suggestion
celebrated with our community a great milestone: in August, Conan Center
```

[reached 1,000 recipes](https://www.youtube.com/watch?v=PjiJ-3sxKbs).

So Conan and Conan Center have come a long way over the last few years. Let's take a look at some
numbers to see how much we have grown.

## Conan

According to the [ISO C++ 2021 survey](https://isocpp.org/files/papers/CppDevSurvey-2021-04-summary.pdf),
Conan is the **most used C/C++ package manager**.

So Conan is the most used, but *how much*? There are
[a lot of ways](https://conan.io/downloads.html) to install Conan, and we don't have data on

> installs coming from mirrors. The most popular source, though, is PyPI, the Python package index.

Reviewer: Maybe I would omit the sentence: "There are
[a lot of ways](https://conan.io/downloads.html) to install Conan, and we don't have data on
installs coming from mirrors" just a personal preference to go straight to the point but it's not important.

Author: The reason for this is to point out that the following stats are coming only from PyPI, not the rest of installers

---

2021 has been quite a busy year for Conan. We are working hard to get Conan 2.0 ready, almost 4
years after 1.0 was released. And, of course, the 1.X series is still very active, with 10
versions released during the year.

And Conan Center has not lagged behind. In 2021, only two years after its inception, we
celebrated with our community a great milestone: on August, Conan Center
[reached 1,000 recipes](https://www.youtube.com/watch?v=PjiJ-3sxKbs).

So Conan and Conan Center have come a long way over the last few years. Let's take a look at some
numbers to see how much we have grown.

## Conan

According to the [ISO C++ 2021 survey](https://isocpp.org/files/papers/CppDevSurvey-2021-04-summary.pdf),
Conan is the **most used C/C++ package manager**.

So Conan is the most used, but *how much*? There are
[a lot of ways](https://conan.io/downloads.html) to install Conan, and we don't have data on
installs coming from mirrors. The most popular source, though, is PyPI, the Python package index.
In there, we are currently having around **500.000 monthly downloads**.
[Almost 520.000 in December](https://pypistats.org/packages/conan).

> But the true indicator of how healthy an Open Source project is, is its GitHub activity. Here are

Reviewer: true --> most valuable, most important... ?
```suggestion
But the true indicator of how healthy an Open Source project is in its GitHub activity. Here are
```

</figure>

The most visited part of our site are the following:

{:class="table table-bordered small"}
| Section                                           | Avg visit time (minutes) |
|---------------------------------------------------|--------------------------|
| [Main site](https://conan.io)                     | 1:16                     | 
| [Main Conan Center site](https://conan.io/center) | 3:52                     |
| [Documentation](https://docs.conan.io)            | 3:15                     |
| [Blog](https://blog.conan.io/)                    | 7:09                     |

It's also worth noting that most of our visits are organic, coming from search engines:

<figure>
    <img src="{{ site.url }}/assets/post_images/2022-01-04/web-channels.png">
</figure>

### Slack

Our channel in the [CppLang slack server](https://cpplang.slack.com/), #conan, is the 2nd most
used channel by members posting and the 4th most active by published messages.

<figure>
    <figcaption>Ranking of CppLang Slack Server Channels:</figcaption>
    <img src="{{ site.url }}/assets/post_images/2022-01-04/slack.png">
</figure>

## Conan Center

> When we created Conan Center in August 2019 we were trying to fill a void in our ecosystem: back 
> then, the latest version of the Conan client was 1.18, so it was completely production-ready, but
> we didn't have our own package repository to handle dependencies in a straightforward way. We 
> wanted for our users to be able to install open source libraries and tools by just adding a couple
> of lines to their ConanFiles.

Reviewer: ```suggestion
When we created Conan Center in August 2019, we were trying to fill a void in our ecosystem: back 
then, the latest version of the Conan client was 1.18, so it was completely production-ready, but
we didn't have our own package repository to handle dependencies straightforwardly. We 
wanted for our users to be able to install open source libraries and tools by just adding a couple
of lines to their ConanFiles.
```

<figure>
    <figcaption>Merged, rejected and open PRs over time:</figcaption>
    <img src="{{ site.url }}/assets/post_images/2022-01-04/pr-by-state.png">
</figure>
<figure>
    <figcaption>Hours before a new PR is merged:</figcaption>
    <img src="{{ site.url }}/assets/post_images/2022-01-04/time-until-closed-pr.png">
</figure>

But Conan Center is not only a recipe repository. We also build and store binaries for the most
common configurations so our users don't have to build their dependencies, which can take A LOT
of time. Those packages are stored in an Artifactory repository. Here are some numbers around it:

{:class="table table-bordered small"}
| **Total number of binaries** | **11.050.000** |
|------------------------------|----------------|
| **Total size of binaries**   | **11.28TB**    |

Of course, having all this in place would be pointless if anybody was using it. So let's take a
look at how much data traffic is going through our remotes:

<figure>
    <figcaption>Daily Data Transfer (GB):</figcaption>
    <img src="{{ site.url }}/assets/post_images/2022-01-04/data-transfer-1.png">
</figure>
<figure>
    <figcaption>Monthly Data Transfer (GB):</figcaption>

>     <img src="{{ site.url }}/assets/post_images/2022-01-04/data-transfer-2.png">

Reviewer: This figure adds conan.bintray.com to the bars, do we want this? Or just showing the conancenter data transfer?

Reviewer 2: Not sure, but at least something related to "Bintray shutdown, no legacy behind", to explain why those numbers go to zero for Bintray. maybe?
